### PR TITLE
Update validation invariant logging

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

File: packages/core/src/fhirpath/parse.ts [hunk 1]
```diff
@@ -259,5 +259,6 @@ export function evalFhirPathTyped(
   variables?: Record<string, TypedValue>
 ): TypedValue[] {
   const variableInput = variables ? variables : {};
-  return parseFhirPath(expression).eval({ variables: variableInput }, input);
+  const ast = parseFhirPath(expression);
```
Comment (mattwiller) [inline packages/core/src/fhirpath/parse.ts:262]: Breaking this out onto its own line was helpful for debugging, to inspect the parsed tree of atoms before evaluating the expression

File: packages/server/src/fhir/repo.ts [hunk 2]
```diff
@@ -530,18 +529,21 @@ export class Repository extends BaseRepository implements FhirRepository {
...
+          logger.error(
+            `New validator invariant error in ${resource.resourceType}/${resource.id}: ${JSON.stringify(
+              invariantErrors
+            )}`
+          );
         }
-        if (structureErrors.length > 0) {
-          throw new OperationOutcomeError({ resourceType: 'OperationOutcome', issue: structureErrors });
+        if (outcome.issue?.length && outcome.issue.length !== invariantErrors?.length) {
+          throw new OperationOutcomeError(outcome);
```
Comment (jamestouri) [inline packages/server/src/fhir/repo.ts:542]: So instead of splitting apart the invariant/non-invariant errors this is going to show all errors?

**Review by jamestouri — APPROVED** [review]

File: packages/server/src/fhir/repo.ts (hunk 2 above)
Comment (mattwiller) [reply packages/server/src/fhir/repo.ts:542]: Yep!  @codyebberson preferred to included them

**Comment by reshmakh** [comment]
Related: https://github.com/medplum/medplum/discussions/2006

**Pull request merged**